### PR TITLE
Actions with require-whisk-auth annotation as boolean false are rejected by controller

### PR DESCRIPTION
**Pull request by nitikagarw**
## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#4985)

## My changes affect the following components

- [ ] API
- [x] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [x] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [ ] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by nitikagarw** [comment]
@rabbah @style95 Please review.

**Review by rabbah — APPROVED** [review]

**Review by style95 — APPROVED** [review]
LGTM
@nitikagarw Thank you for the changes :)

File: tests/src/test/scala/org/apache/openwhisk/core/cli/test/WskWebActionsTests.scala [hunk 1]
```diff
@@ -147,6 +147,25 @@ class WskWebActionsTests extends TestHelpers with WskTestHelpers with RestUtil w
...
+      val host = getServiceURL()
+      val url = s"$host$testRoutePath/$namespace/default/$name.json"
+
+      assetHelper.withCleaner(wsk.action, name) { (action, _) =>
+        action.create(name, file, web = Some("true"), annotations = Map("require-whisk-auth" -> false.toJson))
+      }
+
+      val unauthorizedResponse = RestAssured.given().config(sslconfig).get(url)
+      unauthorizedResponse.statusCode shouldBe 200
+      unauthorizedResponse.body.asString.parseJson.asJsObject.fields("__ow_user").convertTo[String] shouldBe namespace
```
Comment (style95) [inline tests/src/test/scala/org/apache/openwhisk/core/cli/test/WskWebActionsTests.scala:166]: Seems the test complains about the absence of this field.

Comment (nitikagarw) [reply tests/src/test/scala/org/apache/openwhisk/core/cli/test/WskWebActionsTests.scala:166]: I think this field is set only in the case of an Authenticated user and this assert should be removed?
Reference:  https://github.com/apache/openwhisk/blob/master/core/routemgmt/getApi/getApi.js#L27

Comment (style95) [reply tests/src/test/scala/org/apache/openwhisk/core/cli/test/WskWebActionsTests.scala]: Yes

**Pull request merged**